src/ResultSet.php

```diff
@@ -39,12 +54,14 @@ public function current()
             $this->advance();
         }
 
-        return $this->cache->current();
+        return $this->isCacheDisabled ? $this->generator->current() : $this->cache->current();
     }
 
     public function next()
     {
-        $this->cache->next();
+        if (! $this->isCacheDisabled) {
+            $this->cache->next();
+        }
 
         if (! $this->cache->valid()) {
```

Comment:
Why don't you check if the cache is disabled here? I mean, there's little reason to even ask it if it's valid since it will never be if unused.

Reply:
Doesn't `$this->cache->valid()` return false anyway if the cache is disabled?

Reply:
Yes, but that's why I've added:
> there's little reason to even ask it if it's valid since it will never be if unused